chpl/chpl-service/src/main/java/gov/healthit/chpl/upload/listing/ListingUploadHandlerUtil.java

```diff
@@ -0,0 +1,394 @@
+            }
+            certResultHeading = convertToCsvRecord(certResultHeadingValues);
+
+            for (CSVRecord dataRecord : dataRecords) {
+                List<String> certResultColumnValues = new ArrayList<String>();
+                for (int i = certResultColumnRange.getMinimumInteger(); i <= certResultColumnRange.getMaximumInteger(); i++) {
+                    certResultColumnValues.add(dataRecord.get(i));
+                }
+                CSVRecord writtenDataRecord = convertToCsvRecord(certResultColumnValues);
+                if (writtenDataRecord != null) {
+                    certResultRows.add(writtenDataRecord);
+                }
+            }
+        } else {
+            throw new ValidationException(msgUtil.getMessage("certResult.upload.invalidRange",
+                    certResultColumnRange.getMinimumInteger(), certResultColumnRange.getMaximumInteger()));
+        }
+
+        certResultRows.add(0, certResultHeading);
+        return certResultRows;
+    }
+
+    private CSVRecord convertToCsvRecord(List<String> values) {
+        CSVFormat csvFormat = CSVFormat.EXCEL.withRecordSeparator(System.lineSeparator())
+                .withQuoteMode(QuoteMode.ALL);
+        CSVRecord csvRecord = null;
+        final StringWriter out = new StringWriter();
+        try (CSVPrinter csvPrinter = new CSVPrinter(out, csvFormat)) {
+            csvPrinter.printRecord(values);
+            String value = out.toString().trim();
+            csvRecord = CSVParser.parse(value, CSVFormat.EXCEL).getRecords().get(0);
+        } catch (IOException ex) {
+            LOGGER.error("Cannot write values as CSVRecord.", ex);
+        }
+        return csvRecord;
+    }
+
+    @SuppressWarnings("checkstyle:magicnumber")
+    public String parseCriteriaNumberFromHeading(String headingVal) {
+        if (StringUtils.isEmpty(headingVal)) {
+            return null;
+        }
+        headingVal = headingVal.trim().toUpperCase();
+        String criterionNumber = null;
+        Matcher m = criterionColHeadingPattern.matcher(headingVal);
+        if (m.find()) {
+            criterionNumber = m.group(1) + "." + m.group(2)
+                + " (" + m.group(3).toLowerCase() + ")(" + m.group(4) + ")";
+            if (m.group(5) != null) {
+                criterionNumber += "(" + m.group(5).toUpperCase() + ")";
+            }
+         }
+        return criterionNumber;
+    }
+
+    public boolean isCriteriaNumberHeadingCures(String headingVal) {
+        return headingVal.toUpperCase().contains(CRITERIA_CURES_COL_HEADING);
+    }
+
+    @SafeVarargs
+    public final boolean areCollectionsEmpty(Collection<String>... collections) {
+        List<Collection<String>> nonEmptyNonNullCollections =
+                Stream.of(collections)
+                    .filter(Objects::nonNull)
+                    .filter(CollectionUtils::isNotEmpty)
+                    .collect(Collectors.toList());
```

Comment:
Nothing wrong here, but wondering why you used the alternate Stream library here?

Reply:
Because the "collections" is an array (varargs in the method signature) and I couldn't just say collections.stream() on that. I guess I could have done Arrays.asList(collections).stream(). Not sure if one is really preferable or if there is something else I'm not thinking of? I suspect I picked the one that showed up first when I googled how to do it. I'm happy to change it though if you think it's better to do it the other even to just match style or something?